pystac_client/collection_client.py

```diff
@@ -23,7 +23,7 @@ def get_items(self) -> Iterable["Item_Type"]:
 
         link = self.get_single_link('items')
         if link is not None:
-            search = ItemSearch(link.href, method='GET')
+            search = ItemSearch(link.href, method='GET', stac_io=self.get_root()._stac_io)
```

Comment:
Nothing technically wrong here, but accessing a "private" member of an object retrieved via a function call feels like we're breaking abstraction. This isn't surprising given the nature of how the `StacIO` stuff has to work, but I'd just flag this for a bit of code smell w/ maybe the chance to refactor sometime down the line. No changes required now.

Reply:
Yeah, thanks for flagging that. I had originally added a "public" `stac_io` property to `pystac.Catalog` in stac-utils/pystac#590, but we ultimately decided that it wasn't needed as part of that PR, so I took it back out. Given that there are a couple of places where it is accessed in the `pystac-client` code, it might be worth revisiting that.

Reply:
Opened stac-utils/pystac#675 to discuss adding a "public" attribute/property.